### ISSUE TITLE
WindowsAuthenticate middleware doesn't bind LDAP users

### ISSUE THREAD

**Issue by munverzagt** [Adldap2/Adldap2#425]
- Adldap2 Version: 3.0.5
- LDAP Type: Active Directory
- Laravel Version: 5.3
- PHP Version: 7.1.7

### Description:
After logout and relogin i get the error "Call to a member function inGroup() on null". After reload i get "You are logged in"-page.

### Steps To Reproduce:
I have the same source code as in this Issue: [Schema is null](https://github.com/Adldap2/Adldap2/issues/423). When is the property ldap set to Adldap\User model? This is my middleware where user is being logged in. Did i miss something?
```php
namespace App\Http\Middleware;

use Closure;
use Adldap\Models\ModelNotFoundException;
use Adldap\Laravel\Traits\UsesAdldap;
use Adldap\Laravel\Traits\DispatchesAuthEvents;
use Adldap\Laravel\Auth\DatabaseUserProvider;
use Adldap\Laravel\Auth\NoDatabaseUserProvider;
use Illuminate\Http\Request;
use Illuminate\Contracts\Auth\Guard;
use Illuminate\Auth\AuthenticationException;

class WindowsAuthenticate
{
    use UsesAdldap, DispatchesAuthEvents;
    /**
     * The authenticator implementation.
     *
     * @var \Illuminate\Contracts\Auth\Guard
     */
    protected $auth;
    /**
     * Create a new filter instance.
     *
     * @param Guard $auth
     */
    public function __construct(Guard $auth)
    {
        $this->auth = $auth;
    }
    /**
     * Handle an incoming request.
     *
     * @param Request $request
     * @param Closure $next
     *
     * @return mixed
     */
    public function handle(Request $request, Closure $next)
    {
        if (!$this->auth->check()) {
            // Retrieve the SSO login attribute.
            $auth = $this->getWindowsAuthAttribute();
            // Retrieve the SSO input key.
            $key = key($auth);

            // Handle Windows Authentication.
            if ($account = $request->server($auth[$key])) {
                // Username's may be prefixed with their domain,
                // we just need their account name.
                $username = explode('\\', $account);
                if (count($username) === 2) {
                    list($domain, $username) = $username;
                } else {
                    $username = $username[key($username)];
                }
                if ($user = $this->retrieveAuthenticatedUser($key, $username)) {
                    $this->auth->login($user);
                }
            }
        }
        return $next($request);
    }
    /**
     * Returns the authenticatable user instance.
     *
     * @param string $key
     * @param string $username
     *
     * @return \Illuminate\Contracts\Auth\Authenticatable|null
     */
    protected function retrieveAuthenticatedUser($key, $username)
    {
        $provider = $this->auth->getProvider();
        try {
            $resolver = $this->getResolver();

            // Find the user in AD.
            $user = $resolver->query()->where([$key => $username])->firstOrFail();
            if ($provider instanceof NoDatabaseUserProvider) {
                $this->handleAuthenticatedWithWindows($user);
                return $user;
            } elseif ($provider instanceof DatabaseUserProvider) {
                $credentials = [
                    $resolver->getEloquentUsername() => $user->getFirstAttribute($resolver->getLdapUsername()),
                ];
                // Here we'll import the AD user. If the user already exists in
                // our local database, it will be returned from the importer.
                $model = $this->getImporter()->run($user, $this->getModel(), $credentials);
                // We'll assign a random password for the authenticating user.
                $password = str_random();
                // Set the models password.
                $model->password = $model->hasSetMutator('password') ?
                    $password : bcrypt($password);
                // We also want to save the returned model in case it doesn't
                // exist yet, or there are changes to be synced.
                $model->save();
                $this->handleAuthenticatedWithWindows($user, $model);
                return $model;
            }
        } catch (ModelNotFoundException $e) {
            throw new AuthenticationException("Zugang verweigert!");
        }
    }
    /**
     * Returns the configured authentication model.
     *
     * @return \Illuminate\Database\Eloquent\Model
     */
    protected function getModel()
    {
        return auth()->getProvider()->createModel();
    }
    /**
     * Returns the windows authentication attribute.
     *
     * @return string
     */
    protected function getWindowsAuthAttribute()
    {
        return config('adldap_auth.windows_auth_attribute', ['samaccountname' => 'AUTH_USER']);
    }
}
```

**Comment by stevebauman** [comment]
Hi @munverzagt, I believe this is actually a bug. This is due to the LDAP user not being bound to the users model after authentication with the built in middleware.

You can see this taking place with the `DatabaseUserProvider` here:

https://github.com/Adldap2/Adldap2-Laravel/blob/master/src/Auth/DatabaseUserProvider.php#L146-L151

Will have this fixed, however need to figure out the correct approach here.

**Comment by jelenpivo** [comment]
AD.
its working fine with DN $app->ad->auth()->attempt('CN=USER,OU=test,OU=testing,DC=example,DC=ex,DC=com','PASS', $bindAsUser = true)
But not $app->ad->auth()->attempt('USER','PASS', $bindAsUser = true)

**Comment by stevebauman** [comment]
This has been fixed in `v4.0`, however I will definitely accept a PR for `v3.0` if requested.

Thanks! :)
